torch/nn/functional.py

```diff
@@ -3343,20 +3346,24 @@ def multi_head_attention_forward(query,                           # type: Tensor
             v = linear(value, v_proj_weight_non_opt, in_proj_bias)
     q = q * scaling
 
+    if attn_mask is not None:
+        if attn_mask.dim() == 2:
+            attn_mask = attn_mask.unsqueeze(0)
+        elif attn_mask.dim() == 3:
+            if list(attn_mask.size()) != [bsz * num_heads, query.size(0), key.size(0)]:
```

Comment:
Is there an equivalent check for the 2d mask? If so, could we move the argument checking into one place?

Reply:
good catch. the check should be true for both cases.

Reply:
good catch. the check should be true for both cases.